sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -92,51 +85,80 @@ static void connect_mqtt_client_to_iot_hub(void);
+// General message building, sending, receiving functions
+static void set_publish_topic(
+    pnp_publish_topic topic,
+    az_span request_id,
+    az_iot_status status,
+    az_span component_name,
+    char* out_topic,
+    size_t topic_length);
+static void publish_mqtt_message(char const* topic, az_span payload, int qos);
+static void receive_mqtt_message(void);
+static void on_message_received(
+    char* topic,
+    int topic_len,
+    MQTTClient_message const* receive_message);
+
+// Device twin, command request, telemetry functions
 
-// Device Twin functions
 static void handle_device_twin_message(
-    const az_span twin_message_span,
-    const az_iot_hub_client_twin_response* twin_response);
-
-// Command functions
-static void handle_command_message(
-    const az_span command_message_span,
-    const az_iot_hub_client_method_request* command_request);
-static az_result temp_controller_process_command(
-    const az_iot_hub_client_method_request* command_request,
+    MQTTClient_message const* receive_message,
+    az_iot_hub_client_twin_response const* twin_response);
+static void handle_command_request(
+    MQTTClient_message const* receive_message,
+    az_iot_hub_client_method_request const* command_request);
+static void send_telemetry_messages(void);
+
+// Temperature controller functions
+static void temp_controller_build_serial_number_reported_property(
+    az_span payload,
+    az_span* out_payload);
+static void temp_controller_build_error_reported_property_with_status(
     az_span component_name,
+    az_span property_name,
+    az_json_reader* const property_value,
+    az_iot_status status,
+    int32_t version,
+    az_span payload,
+    az_span* out_payload);
+static void temp_controller_build_telemetry_message(az_span payload, az_span* out_payload);
+static az_result temp_controller_process_command_request(
     az_span command_name,
     az_span command_payload,
-    pnp_mqtt_message* mqtt_message,
-    az_iot_status* status);
-
-// Telemetry functions
-static void send_telemetry_messages(void);
-static az_result temp_controller_get_telemetry_message(pnp_mqtt_message* message);
+    az_span payload,
+    az_span* out_payload,
+    az_iot_status* out_status);
+static az_result temp_controller_process_property_update(
+    az_span component_name,
+    az_json_token const* property_name,
+    az_json_reader const* const property_value,
+    int32_t version,
+    az_span payload,
+    az_span* out_payload);
+static void temp_controller_invoke_reboot(void);
 
 // Callbacks
-static az_result append_string_callback(az_json_writer* jw, void* value);
-static az_result append_json_token_callback(az_json_writer* jw, void* value);
 static void property_callback(
     az_span component_name,
-    const az_json_token* property_name,
-    az_json_reader property_value,
+    az_json_token const* property_name,
+    az_json_reader* const property_value,
```

Comment:
From what I remember, we have tabled whether we add `const` on pointers. I would prefer not adding this because you might have to individually change it back.

Reply:
I think that convo was more general in nature.  You can still have instances where you specifically want an address to be const.  This, however, was from my misunderstanding of an earlier comment you wrote.  I have changed it back.